brats/data/datasets.py

```diff
@@ -18,15 +19,20 @@ class NiftiFolder(data.Dataset):
     (Note that the order of the modalities doesn't matter, however it should be consistent for whole dataset)
     """
 
-    def __init__(self, root: str, transform=None):
-        self._files = list(os.scandir(root))
+    def __init__(self, paths: typing.List[str], transform: typing.List[typing.Callable] = None):
+        self._files = paths
         self._transform = transform
 
+    @classmethod
+    def from_dir(cls, root: str, transforms: typing.Callable = None):
```

Comment:
What is the benefit of this appraoch? Why not just perform it in the constructor? Also, I can see that this aproach is used only in tests, not in the actual usage (the `main` function)

Reply:
I changed it this way because the paths to files are given in the form of json file, which is more rror prone than just reading the directory. Still, reading from directory would be useful in a number of scenarios, so I added it this way. I modified the training script so it uses it.